uniffi_core/src/ffi/rustcalls.rs

```diff
@@ -171,11 +161,7 @@ where
         // Callback returned an Err.
         Ok(Err(buf)) => {
             out_status.code = RustCallStatusCode::Error;
-            unsafe {
-                // Unsafe because we're setting the `MaybeUninit` value, see above for safety
-                // invariants.
-                out_status.error_buf.as_mut_ptr().write(buf);
-            }
+            *out_status.error_buf = buf;
```

Comment:
If we set this field multiple times, then we'll leak any previous values.  Can you add a comment saying that we won't leak because a) it's only set in this function and b) there's no branch where it gets set twice.